File: src/beaverdash/ProjectManagementService/src/PM.Domain/Entities/Team.cs
namespace PM.Domain.Entities;

public class Team
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    
    public Guid OwnerUserId { get; set; }
    public User? OwnerUser { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<TeamMember> Members { get; set; } = new List<TeamMember>();
    public ICollection<Project> Projects { get; set; } = new List<Project>();
}

File: src/beaverdash/web/src/types/team.ts
import { User } from "./auth";
import { Project } from "./project";

/**
 * Team interface mapped from PM.Domain.Entities.Team
 */
export interface Team {
  id: string; // UUID
  name: string;
  description: string | null;
  ownerUserId: string; // UUID
  ownerUser?: User | null;
  createdAt: string; // ISO Date String
  updatedAt: string; // ISO Date String
  members?: TeamMember[];
  projects?: Project[];
}

/**
 * TeamMember interface mapped from PM.Domain.Entities.TeamMember
 */
export interface TeamMember {
  teamId: string; // UUID
  team?: Team | null;
  userId: string; // UUID
  user?: User | null;
  role: "Owner" | "Member"; // "Owner" = Trưởng nhóm, "Member" = Thành viên
  joinedAt: string; // ISO Date String
}
